pkg/controllers/deprovisioning/controller.go

```diff
@@ -107,15 +110,22 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) controller.Bu
 }
 
 func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
```

Comment:
WDYT about also logging once per deprovisioning loop at the end if no deprovisoners execute. 

e.g.
Line 142:
```
logging.FromContext(ctx).Debug("Deprovisioning finished, nothing to do")
```